### PR TITLE
ci: add angular robot config file

### PR DESCRIPTION
**Pull request by ocombe**
## PR Type
What kind of change does this PR introduce?

```
[x] CI related changes
```

## What is the current behavior?
No config file for the Angular Robot

## What is the new behavior?
Config file for the Angular Robot

## Does this PR introduce a breaking change?
```
[x] No
```

### CODE REVIEW

File: .github/angular-robot.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - "continuous-integration/travis-ci/pr"
+      - "code-review/pullapprove"
+      - "ci/circleci: build"
+      - "ci/circleci: lint"
+
+  # the comment that will be added when the merge label is removed, leave empty or set to false to disable
+  # {{MERGE_LABEL}} will be replaced by the value of the mergeLabel option
+  # {{OVERRIDE_LABEL}} will be replaced by the value of the overrideLabel option
+  # {{PLACEHOLDER}} will be replaced by the list of failing checks
+  mergeRemovedComment: "I don't like to brag, but I just saved you from a horrible, slow and painful death by removing the `{{MERGE_LABEL}}` label. Probably. Maybe...
```
Comment (IgorMinar) [inline .github/angular-robot.yml]: "saved you" -> "saved this PR"

File: .github/angular-robot.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+\nThe following checks are failing:
+\n{{PLACEHOLDER}}
+\n
+\nBut if you think that you know better than me, then please, go ahead, add the `{{OVERRIDE_LABEL}}` label and add an override justification comment for the caretaker. You'll be free to do whatever you want. Don't say that I didn't warn you."
```
Comment (IgorMinar) [inline .github/angular-robot.yml]: we are removing the override, please update this accordingly.

File: docs/TRIAGE_AND_LABELS.md [hunk 3]
```diff
@@ -114,7 +114,7 @@ Every triaged PR must have a `pr_action` label assigned to it:
 * `PR action: review` - work is complete and comment is needed from the reviewers.
 * `PR action: cleanup` - more work is needed from the author.
 * `PR action: discuss` - discussion is needed, to be led by the author.
-* `PR action: merge` - the PR is ready to be merged by the caretaker.
+* `PR action: merge` - the PR is ready to be merged by the caretaker. If the Angular Robot is removing this label, you can add the `override` label to bypass it. See [Special labels](#override) below.
```
Comment (IgorMinar) [inline docs/TRIAGE_AND_LABELS.md]: undo

File: docs/TRIAGE_AND_LABELS.md [hunk 4]
```diff
@@ -160,3 +160,6 @@ Only issues with `cla:yes` should be merged into master.
 
 ### `aio: preview`
 Applying this label to a PR makes the angular.io preview available regardless of the author. [More info](../aio/aio-builds-setup/docs/overview--security-model.md)
+
+### `override`
```
Comment (IgorMinar) [inline docs/TRIAGE_AND_LABELS.md]: undo

File: .github/angular-robot.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    disabled: false
+    # the name of the status
+    context: "ci/angular: merge status"
+    # text to show when all checks pass
+    successText: "All checks passed!"
+    # text to show when some checks are failing
+    failureText: "The following checks are failing:"
+
+  # comment that will be added to a PR when there is a conflict, leave empty or set to false to disable
+  mergeConflictComment: "Hello? Don't want to hassle you. Sure you're busy. But--this PR has some conflicts that you probably ought to resolve.
```
Comment (IgorMinar) [inline .github/angular-robot.yml]: "But--this" -> "But this"

Comment (IgorMinar) [inline .github/angular-robot.yml]: conflicts -> merge conflicts

File: .github/angular-robot.yml (hunk 2 above)
Comment (IgorMinar) [reply .github/angular-robot.yml]: replace with something like "Please resolve these issues to make it eligible for merging"

File: .github/angular-robot.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    # list of PR statuses that need to be successful
+    requiredStatuses:
+      - "continuous-integration/travis-ci/pr"
+      - "code-review/pullapprove"
+      - "ci/circleci: build"
+      - "ci/circleci: lint"
+
+  # the comment that will be added when the merge label is removed, leave empty or set to false to disable
+  # {{MERGE_LABEL}} will be replaced by the value of the mergeLabel option
+  # {{OVERRIDE_LABEL}} will be replaced by the value of the overrideLabel option
```
Comment (IgorMinar) [inline .github/angular-robot.yml]: remove

**Review by IgorMinar — CHANGES_REQUESTED** [review]

**Comment by ocombe** [comment]
Ok PR updated!

File: .github/angular-robot.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    disabled: false
+    # the name of the status
+    context: "ci/angular: merge status"
+    # text to show when all checks pass
+    successText: "All checks passed!"
+    # text to show when some checks are failing
+    failureText: "The following checks are failing:"
+
+  # comment that will be added to a PR when there is a conflict, leave empty or set to false to disable
+  mergeConflictComment: "Hello? Don't want to hassle you. Sure you're busy. But this PR has some conflicts that you probably ought to resolve.
```
Comment (IgorMinar) [inline .github/angular-robot.yml]: "conflicts" => "merge conflicts"

File: .github/angular-robot.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      - "ci/circleci: build"
+      - "ci/circleci: lint"
+
+    # the comment that will be added when the merge label is added despite failing checks, leave empty or set to false to disable
+    # {{MERGE_LABEL}} will be replaced by the value of the mergeLabel option
+    # {{PLACEHOLDER}} will be replaced by the list of failing checks
+    mergeRemovedComment: "I see that you just added the `{{MERGE_LABEL}}` label. It won't do anything good though, because the following checks are still failing:
+  \n{{PLACEHOLDER}}
+  \n
+  \nIf you want your PR to be merged, it has to pass all the checks. But if you have a good reason to want to merge this, please contact the caretaker to let them know."
```
Comment (IgorMinar) [inline .github/angular-robot.yml]: "If you want your PR to be merged, it has to pass all the checks. If you can't get the PR green due to flakes, please try restarting the CI job. If that fails and you believe that the issue is not due to your change, please contact the caretaker and ask for help."

**Review by IgorMinar — APPROVED** [review]
a few small changes, otherwise lgtm

**Review by IgorMinar — APPROVED** [review]

**Comment by IgorMinar** [comment]
I fixed the issued and forcepushed into this branch

**Pull request closed without merging**